scripts/generate_resource.sh

```diff
@@ -0,0 +1,238 @@
+#!/bin/bash
+
+set -euo pipefail
+
+info() {
+  printf "\r\033[00;35m$1\033[0m\n"
+}
+
+success() {
+  printf "\r\033[00;32m$1\033[0m\n"
+}
+
+fail() {
+  printf "\r\033[0;31m$1\033[0m\n"
+}
+
+__help="This script is used to quickly scaffold a resource in go-tfe. Simply provide a resource name as the first argument and it will generate standard boilerplate.
+\n- The resource name cannot be plural.
+\n- The resource name cannot contain numbers or special characters.
+\nNote: This script cannot generate appropriate camelcasing for resource names.\n
+Example usage: ./generate_resource.sh Organization
+"
+
+# Ensure a resource name is passed as an argument
+if [[ "$#" -ne 1 ]]; then
+    fail "Invalid number of arguments passed. Must specify a single resource name."
+    exit 1
+fi
+
+arg="$1"
+if [[ $arg = "-h" ]]; then
+    echo -e $__help
+    exit 1
+fi
+
+
+# Ensure the resource name isn't plural
+if [[ "${arg: -1}" = "s" ]]; then
+    fail "Error: Resource name cannot be plural."
+    exit 1
+fi
+
+if [[ "${arg}" =~ [^a-zA-Z] ]]; then
+    fail "Error: Resource name can only contain letters."
+    exit 1
+fi
```

Comment:
I tried regenerating a resource that already exist and the existing files get rewritten. Is it possible to check if the resource already exists and inform the user? Or should it rewrite the file and leave it to the user to undo? 🤔